src/ModelContextProtocol/Protocol/Transport/SseClientTransport.cs

```diff
@@ -51,6 +52,32 @@ public SseClientTransport(SseClientTransportOptions transportOptions, HttpClient
         Name = transportOptions.Name ?? transportOptions.Endpoint.ToString();
     }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SseClientTransport"/> class with authentication support.
+    /// </summary>
+    /// <param name="transportOptions">Configuration options for the transport.</param>
+    /// <param name="authorizationProvider">The authorization provider to use for authentication.</param>
+    /// <param name="loggerFactory">Logger factory for creating loggers used for diagnostic output during transport operations.</param>
+    public SseClientTransport(SseClientTransportOptions transportOptions, IMcpAuthorizationProvider authorizationProvider, ILoggerFactory? loggerFactory = null)
+    {
+        Throw.IfNull(transportOptions);
+        Throw.IfNull(authorizationProvider);
+
+        _options = transportOptions;
+        _loggerFactory = loggerFactory;
+        Name = transportOptions.Name ?? transportOptions.Endpoint.ToString();
+
+        // Create an auth handler with the authorization provider
+        var authHandler = new AuthorizationDelegatingHandler(authorizationProvider)
+        {
+            InnerHandler = new HttpClientHandler()
+        };
```

Comment:
How does someone use SseClientTransport with both the IMcpAuthorizationProvider and the rest of their HTTP client pipeline they might want to use, e.g. resilience handlers, logging handlers, etc.? Do we instead need to compose nicely with DI mechanisms here around HttpClientFactory?

Reply:
@stephentoub I updated the constructor to accept a custom `HttpMessageHandler` - would that be sufficient in this scenario? It's then injected into the delegating handler:

```csharp
InnerHandler = baseMessageHandler ?? new HttpClientHandler()
```

Reply:
It's sufficient _iff_ the auth handler always needs to be the first in the pipeline. Does it?

cc: @halter73

Reply:
Does this need to be a DelegatingHandler? In https://github.com/modelcontextprotocol/csharp-sdk/pull/377#discussion_r2080787775 I recommend handling 401 Unauthorized HttpResponseMessages in the SseClientTransport.

I kind of see the appeal of then being able to use the access token you obtained for the MCP server for other HTTP requests, but I would rather leave developers to their existing mechanisms for obtaining tokens for non-MCP servers. I don't think we should be creating a general-purpose client auth library.